### PR TITLE
:bug: Fix copy and paste components between files

### PR DESCRIPTION
**Pull request by pabloalba**
Implements https://tree.taiga.io/project/penpot/task/5111

### CODE REVIEW

File: frontend/src/app/main/data/workspace/libraries_helpers.cljs [hunk 1]
```diff
@@ -61,28 +61,9 @@
...
-            (prepare-create-group it            ; This function needs to be passed as argument
-                                  objects       ; to avoid a circular dependence
-                                  page-id
-                                  shapes
-                                  group-name
-                                  (not (ctk/instance-root? (first shapes))))))
-
+(defn generate-add-component-changes
+  [changes group objects file-id page-id components-v2]
+  (let [_ (prn "generate-add-component-changes" components-v2)
```
Comment (hirunatan) [inline frontend/src/app/main/data/workspace/libraries_helpers.cljs:66]: prn

File: frontend/src/app/main/data/workspace/selection.cljs [hunk 2]
```diff
@@ -359,88 +361,113 @@
...
+;; Return true if the object is a component that exists on the file or its libraries (even a deleted one)
+(defn- is-known-component?
+  [obj libraries library-data]
+  (let [main-instance?    (ctk/main-instance? obj)
+        component-id      (:component-id obj)
+        file-id           (:component-file obj)
+        main-component    (ctf/get-component libraries file-id component-id)
+        deleted-component (ctkl/get-deleted-component library-data component-id)]
+    (and main-instance?
+         (or main-component deleted-component))))
```
Comment (hirunatan) [inline frontend/src/app/main/data/workspace/selection.cljs:408]: This probably should be a helper in `common.types.file`

Comment (hirunatan) [reply frontend/src/app/main/data/workspace/selection.cljs:408]: or even better, add an `include-deleted?` option to `ctf/get-component`, thus avoiding to have two calls

File: frontend/src/app/main/data/workspace/selection.cljs [hunk 3]
```diff
@@ -359,88 +361,113 @@
...
+
+       (is-known-component? obj libraries library-data)
+       (prepare-duplicate-component-change changes page obj delta libraries library-data it)
+
+       :else
+       (let [frame?             (cph/frame-shape? obj)
+             new-id             (ids-map (:id obj))
+             parent-id          (or parent-id frame-id)
+             name               (:name obj)
+             is-component-root? (:foreign-component-root? obj)
```
Comment (hirunatan) [inline frontend/src/app/main/data/workspace/selection.cljs:427]: Is this flag really necessary? if obj is a component root and we are here, only could be if the component is defined in an external file that is not being used by the current one.

Comment (pabloalba) [reply frontend/src/app/main/data/workspace/selection.cljs:427]: Yes, but there is a cleanup of :component-root for external components in frontend/src/app/main/data/workspace.cljs, so I added there the flag to not lose the info

File: frontend/src/app/main/data/workspace/selection.cljs (hunk 2 above)
Comment (pabloalba) [reply frontend/src/app/main/data/workspace/selection.cljs:408]: Fixed on new PR

File: frontend/src/app/main/data/workspace/libraries_helpers.cljs (hunk 1 above)
Comment (pabloalba) [reply frontend/src/app/main/data/workspace/libraries_helpers.cljs:66]: fixed on new PR

**Pull request closed without merging**